### DISCUSSION TITLE
Shroomloin changes

### DISCUSSION THREAD

**Discussion by neoNoteCoder** [infernalstudios/Infernal-Expansion#257] [Ideas]
**Describe the addition:**
After experience from nether survival and regular gameplay, I feel as if the shroomloins are way too aggressive even for the crimson forest and should be nerfed slightly.

The crimson forest is a dense area due to the trees surrounding the roof of the nether. This makes it difficult to predict if shroomloins are around and may be alerted. Not only is the generation an issue for the shroomloin's functionality, but their detection range is majorly unfair. From experience, the shroomloins may sometimes drop down from a higher ledge when you break a stem or vine. This lets the player collect the ascus bombs for free instead of a player actually challenging them for the reward. Testing its range shows that it can detect a broken vine or stem block up to 30 blocks even with block obstructions. This makes the mobs incredibly irritating to fight when gathering wood, especially if they are on a higher ledge where they can shoot further.

Another issue is how far you need to run from the shroomloins so that they stop chasing you. Even when you leave the biome after irritating them, you get punished further for seemingly no reason. And with the amount that chases a player for breaking one block, along with their tendency to head bash, they become a monstrous case to deal with at all at once.

To create a balance for the shroomloin, I suggest the following changes:
- Change their detection range when you break a block _(bc dear lord that's the most annoying part about them)_
- Every block you break that irritates the shroomloin makes **one** shroom angry (To avoid too many shroomloins attacking at once.)
- Weeping vines shouldn't anger shroomloins (Vines are easy to break and can be annoying as a consequence, since a misclick is all that is needed.)
- Shroomloins should not chase you from outside the biome (They should keep their aggressive chase range, just not using it to reach the player from outside their dedicated biome.

**Additional Context:**
There is a bit of an inconsistency with which blocks or items anger a shroomloin. Crimson Stems don't anger the shroomloin, but Crimson Fungus Caps do.
Also, being able to change shroomloins with different mushrooms doesn't fix this issue. Ones that are visible to the player can prevent their anger, but more can still show up thanks to the detection range. They will also still be angry if you change them after breaking the block.

**Comment by Hellion-Games** [comment]
Ok, here I go. The first point, I can get behind. I see how that could easily get annoying. The second point, not as much. I feel that if we reduce the detection range on breaking things we shouldn't _also_ reduce the amount of Shroomloin that get angry. It'll nerf them a bit too much. Your third point will group in with your additional context. I think we should take a look at what blocks anger the Shroomloins and reevaluate it. When I wrote the code for breaking blocks angering them, it was only the small Crimson Fungus things that angered them and that felt fair. I'm not sure who made the change, but I think that needs to be reevaluated. As for your last point, it just doesn't make sense. No other mob stops chasing you when you leave its native biome. Endermen will chase you out of the Warped Forest, Strays and Husks will chase you wherever. The solution, instead, would be to simply reduce the follow range of Shroomloin. That said, they currently use the same follow range as any other Mob, such as Zombies or Skeletons, so there is no reason that should need a reduction. I will visit a couple of these issues with the team and make some decisions about changes and let you know! Thank you for letting us know of your concern!
